File: app/src/main/java/io/github/romangolovanov/apps/ametro/ui/loaders/ExtendedMapInfo.java
package io.github.romangolovanov.apps.ametro.ui.loaders;

import io.github.romangolovanov.apps.ametro.catalog.entities.MapInfo;

public class ExtendedMapInfo extends MapInfo {

    private ExtendedMapStatus status;
    private boolean selected;

    public ExtendedMapInfo(MapInfo map, ExtendedMapStatus status) {
        super(map);
        this.status = status;
    }

    public ExtendedMapStatus getStatus() {
        return status;
    }

    public boolean isSelected() {
        return selected;
    }

    public void setSelected(boolean selected) {
        this.selected = selected;
    }
}

File: app/src/main/java/io/github/romangolovanov/apps/ametro/ui/loaders/ExtendedMapInfo.kt
package io.github.romangolovanov.apps.ametro.ui.loaders

import io.github.romangolovanov.apps.ametro.catalog.entities.MapInfo

class ExtendedMapInfo(map: MapInfo, val status: ExtendedMapStatus) : MapInfo(map) {
    var selected: Boolean = false
}
